chore(docs): Add deploy to Azure guide (#24230)
gatsbyjs/gatsby: docs/docs/deploying-to-azure.md

1. Enter the **Static Web Apps details**.

   | Setting  | Value                                                                         |
   | -------- | ----------------------------------------------------------------------------- |
   | _Name_   | Name your app. Valid characters are `a-z` (case insensitive), `0-9`, and `_`. |
   | _Region_ | Select Region closest to you                                                  |
   | _SKU_    | **Free**                                                                      |

1. Click the **Sign-in with GitHub** button and authenticate with GitHub.
1. Enter the **Source Control Details**.

   | Setting        | Value                                                    |
   | -------------- | -------------------------------------------------------- |
   | _Organization_ | Select the Organization where you created the repository |
   | _Repository_   | **gatsby-app**                                           |
   | _Branch_       | **master**                                               |

1. Click the **Next: Build >** button to edit the build configuration.

### Building your site

> Next, add configuration details specific to your preferred frontend framework.

Reviewer: Since we're assuming the reader is using Gatsby, this should be spelled out as well as the values below.

| Setting                 | Value         |
| ----------------------- | ------------- |
| _App location_          | _Leave blank_ |

> | _Api location_          | _Leave blank_ |

Reviewer: These are `/` and `api` by default and "App location" is a required field. I think a little more explanation here would be helpful to readers but as long as they know they can leave the default values there, we're fine.

> | _App artifact location_ | **build**     |

Reviewer: ```suggestion
| _App artifact location_ | **public**     |
```

Click the **Review + create** button.

Continue to create the application.

1. Click the **Create** button

1. Once the deployment is complete, click the **Go to resource** button

### Review the GitHub Action

At this stage, your Static Web Apps instance is created in Azure but your app is not yet deployed. The GitHub Action that Azure creates in your repository will run automatically to perform the first build and deployment of your app, but it takes a couple of minutes to finish.

> You can check the status of your build and deploy action by clicking a big blue border at the top of your main window.

Reviewer: ```suggestion
You can check the status of your build and deploy action by clicking the blue information banner at the top of your main window.
```

This could still be misread as referring to the main navbar at the top of the screen.

### View your website

Once your GitHub Action finishes building and publishing your web app, you can browse to see your running app.

Click on the _URL_ link in the Azure portal to visit your app in the browser.

Congratulations! You've deployed your first app to Azure Static Web Apps!

> Don't worry if you see a web page that says the app hasn't been built and deployed yet. Try refreshing the browser in a minute. The GitHub Action runs automatically when the Azure Static Web Apps is created. So if you see the splash page, the app is still being deployed.

## Additional resources

> There's much more to learn about Azure Static Web Apps such as working with routes, setting up custom domains, adding a Serverless API, and much more. Below are some useful links:

Reviewer: These are things which could potentially be added to this reference guide in the future!

Author: These are an important part for people to learn more about the service. Deploying is just the beginning. :)

Author: but if you mean adding written sub-sections on what the links are pointing to then I definitely agree it could be a great thing to add in the future :)

Reviewer: ^ that is exactly what I mean! There's room for this guide to expand. :D

   | _Repository_   | **gatsby-app**                                           |
   | _Branch_       | **master**                                               |

1. Click the **Next: Build >** button to edit the build configuration.

### Building your site

Next, add configuration details specific to your Gatsby project.

The `App location` is the location of your front end app code. If you have a frontend app and an API in different directories but in the same repository, you will need to point out under which sub directory your front end app lives. You can leave this blank if you have a Gatsby project at the root.

The default value for `Api location` is `/api`. As we don't have an API in an `/api` directory we can leave this blank and the Static Web App Service will ignore it during build. If you wish you can add an API at a later stage. Refer to the links at the bottom of this doc that describes how to do just that.

The really important field to specify is the `App artifact location`, this should point to the folder where your Gatsby project is built, this would be `public` directory.

| Setting                 | Value         |
| ----------------------- | ------------- |

> | _App location_          | _Leave blank_ |

Reviewer: ```suggestion
| _App location_          | / |
```

> | _Api location_          | _Leave blank_ |

Reviewer: ```suggestion
| _Api location_          | **api** or _Leave blank_ |
```
